### PR TITLE
Describe how MIMI protocol interacts with MIMI DS

### PR DESCRIPTION
**Pull request by kkohbrok**

### CODE REVIEW

**Review by rohan-wire — APPROVED** [review]
I still have issues with the text, but it is substantially better.

One suggestion, is we rename the "user" in `m.room.user` to "invite"

File: draft-ralston-mimi-protocol.md [hunk 1]
```diff
@@ -181,10 +185,144 @@ by a consistent identifier.
...
+> atomically processable message.
+
+### DSEvents
+
+All events that pertain to the MLS group that underlies a room are wrapped into
+a DSEvent and sent as an Event of type `ds.group`.
+
+~~~ tls
+struct {
+   DSRequest ds_request;
```
Comment (turt2live) [inline draft-ralston-mimi-protocol.md:262]: From an API design perspective, ideally we split `DSRequest` out to individual events.

File: draft-ralston-mimi-protocol.md [hunk 2]
```diff
@@ -181,10 +185,144 @@ by a consistent identifier.
...
+struct {
+  opaque room_id;
+  UserData participants<V>;
+  // TODO: Add any remaining room data
+} RoomState;
+~~~
+
+Whenever a client creates a commit, it includes an UpdateRoom proposal that
+contains all events since the preceeding commit and updates the ParticipantData
+extension accordingly as part of the commit.
```
Comment (turt2live) [inline draft-ralston-mimi-protocol.md:237]: All events sounds like a lot. Do we just need specific event types here? `m.room.user` primarily.

**Review by turt2live — COMMENTED** [review]
from a first pass of review

File: draft-ralston-mimi-protocol.md (hunk 1 above)
Comment (kkohbrok) [reply draft-ralston-mimi-protocol.md:262]: Not a strong opinion, but I thought this was a good way of maintaining the abstraction of a cryptographic building block.

File: draft-ralston-mimi-protocol.md (hunk 2 above)
Comment (kkohbrok) [reply draft-ralston-mimi-protocol.md:237]: It should probably be "all events that change the room state", which currently is only m.room.user.

**Pull request merged**